packages/sdk/test/integration/client_test.ts

```diff
@@ -863,4 +863,137 @@ describe.sequential('Client', function () {
       assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
     }, task.name);
   });
+
+  it('Should successfully broadcast serializeable payload', async ({
+    task,
+  }) => {
+    const cli = new yorkie.Client(testRPCAddr);
+    await cli.activate();
+
+    const doc = new yorkie.Document<{ t: Text }>(toDocKey(`${task.name}`));
+    await cli.attach(doc);
+
+    const broadcastTopic = 'test';
+    const payload = { a: 1, b: '2' };
+
+    expect(async () => doc.broadcast(broadcastTopic, payload)).not.toThrow();
+
+    await cli.deactivate();
+  });
+
+  it('Should throw error when broadcasting unserializeable payload', async ({
+    task,
+  }) => {
+    const cli = new yorkie.Client(testRPCAddr);
+    await cli.activate();
+
+    const doc = new yorkie.Document<{ t: Text }>(toDocKey(`${task.name}`));
+    await cli.attach(doc);
+
+    // broadcast unserializable payload
+    const payload = () => {};
+    const broadcastTopic = 'test';
+
+    expect(async () =>
+      doc.broadcast(broadcastTopic, payload),
+    ).rejects.toThrowErrorCode(Code.ErrInvalidArgument);
+
+    await cli.deactivate();
+  });
+});
+
+it('Should trigger the handler for a subscribed broadcast event', async ({
+  task,
+}) => {
+  await withTwoClientsAndDocuments<{ t: Text }>(
+    async (c1, d1, c2, d2) => {
+      const spy = vi.fn();
+
+      const broadcastTopic = 'test';
+      const unsubscribe = d2.subscribe(
+        { type: 'broadcast', topic: broadcastTopic },
+        spy,
+      );
+
+      const payload = { a: 1, b: '2' };
+      await d1.broadcast(broadcastTopic, payload);
+
+      // Assuming that every subscriber can receive the broadcast event within 1000ms.
+      await new Promise((res) => setTimeout(res, 1000));
```

Comment:
@chacha912 
Thank you for the suggestion. While `EventCollector` is a great alternative to timeouts and I've successfully applied it in several test cases, some scenarios were not well-suited for its use.

For example, in the test case below, the handler won't be called because it hasn't subscribed to the topic. As a result, using `EventCollector` wouldn't be effective here:

```javascript
it('Should not trigger the handler for an unsubscribed broadcast event', async ({
  task,
}) => {
  await withTwoClientsAndDocuments<{ t: Text }>(
    async (c1, d1, c2, d2) => {
      const spy = vi.fn();

      const broadcastTopic1 = 'test1';
      const broadcastTopic2 = 'test2';

      const unsubscribe = d2.subscribe(
        { type: 'broadcast', topic: broadcastTopic2 },
        () => {
           // wont'be called
           spy()
        },
      );

      const payload = { a: 1, b: '2' };
      await d1.broadcast(broadcastTopic1, payload);

      // Assuming that every subscriber can receive the broadcast event within 1000ms.
      await new Promise((res) => setTimeout(res, 1000));

      expect(spy.mock.calls.length).toBe(0);

      unsubscribe();
    },
    task.name,
    SyncMode.Realtime,
  );
});
```

Reply:
While we can use `eventCollector.getLength()` to check if no events have occurred, it seems we might need an additional method to test that no events occur during a timeout period. For now, I think it's fine to proceed with the approach you've suggested. Thank you for making those modifications.